src/tools/compiletest/src/runtest.rs

```diff
@@ -214,10 +215,10 @@ impl<'test> TestCx<'test> {
             self.fatal_proc_rec("compilation failed!", &proc_res);
         }
 
+        // FIXME(#41968): Remove this check after #41968 is merged.
         let expected_errors = errors::load_errors(&self.testpaths.file, self.revision);
-        if !expected_errors.is_empty() {
-            self.check_expected_errors(expected_errors, &proc_res);
-        }
+        assert!(expected_errors.is_empty(),
+                "run-pass tests with expected warnings should be moved to ui-run/");
```

Comment:
@Mark-Simulacrum Well this check does add a tiny overhead to the test 😄 (running `load_errors` will scan the whole file again for `//~`).

(This can be moved to `tidy` but it will spend the same amount of Travis time)